editor/src/messages/layout/layout_message_handler.rs

```diff
@@ -24,14 +25,44 @@ impl<F: Fn(&MessageDiscriminant) -> Vec<KeysGroup>> MessageHandler<LayoutMessage
 		use LayoutMessage::*;
 		#[remain::sorted]
 		match message {
-			RefreshLayout { layout_target } => {
-				self.send_layout(layout_target, responses, &action_input_mapping);
-			}
-			SendLayout { layout, layout_target } => {
-				self.layouts[layout_target as usize] = layout;
-
-				self.send_layout(layout_target, responses, &action_input_mapping);
+			RefreshLayout { layout_target, dirty_id } => {
```

Comment:
I think this should be broken up into two functions. I think a function should be added to get a path from a layout given a widget_id that is impl'd onto the struct. The function signature could be

```rust
fn get_path(&self, widget_id:u64) -> Option<(Path, &Widget)>
```